### PR TITLE
RangeHashed dictionary support dictHas function

### LINKED ISSUE #6680: dictHas is not supported with range_hashed
dictGet works
```
SELECT dictGet('d_test', 'atr', toUInt64(1), toDateTime('2019-02-01 00:00:00')) AS x
┌─x─┐
│ 1 │
└───┘
```

but dictHas does not.
```
SELECT dictHas('d_test', toUInt64(1), toDateTime('2019-02-01 00:00:00')) AS x

Received exception from server (version 19.14.1):
Code: 42. DB::Exception: Received from localhost:9000. 
DB::Exception: Number of arguments for function dictHas doesn't match: passed 3, should be 2.
```

```


CREATE VIEW default.v_test (`id` UInt8, `start` DateTime, `end` DateTime, `atr` UInt8) AS SELECT 1 AS id, toDateTime('2019-01-01 00:00:00') AS start, toDateTime('2020-01-01 00:00:00') AS end, 1 AS atr

<?xml version="1.0"?>
<dictionaries>
<dictionary>
        <name>d_test</name>
        <source>
            <clickhouse>
                <host>localhost</host>
                <port>9000</port>
                <user>default</user>
                <password></password>
                <db>default</db>
                <table>v_test</table>
            </clickhouse>
        </source>

        <layout>
            <range_hashed />
        </layout>

        <structure>
            <id>
                <name>id</name>
            </id>
            <range_min>
                    <name>start</name>
		           <type>DateTime</type>
            </range_min>
            <range_max>
		    <name>end</name>
		    <type>DateTime</type>
            </range_max>
            <attribute>
                    <name>atr</name>
                    <type>UInt8</type>
                    <null_value>0</null_value>
            </attribute>
        </structure>

        <lifetime>
            <min>3000</min>
            <max>3600</max>
        </lifetime>
    </dictionary>
```

### PR DESCRIPTION
**Pull request by kitaisreal**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Fixes #6680.

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Support `dictHas` function for `RangeHashedDictionary`. Fixes #6680.

### CODE REVIEW

File: src/Dictionaries/RangeHashedDictionary.cpp [hunk 1]
```diff
@@ -178,10 +177,76 @@ ColumnPtr RangeHashedDictionary::getColumn(
...
-ColumnUInt8::Ptr RangeHashedDictionary::hasKeys(const Columns &, const DataTypes &) const
+ColumnUInt8::Ptr RangeHashedDictionary::hasKeys(const Columns & key_columns, const DataTypes & key_types) const
 {
-    throw Exception(ErrorCodes::NOT_IMPLEMENTED,
-        "Has not supported", getDictionaryID().getNameForLogs());
+    auto range_storage_column = key_columns[1];
+    ColumnWithTypeAndName column_to_cast = {range_storage_column->convertToFullColumnIfConst(), key_types[1], ""};
+
+    auto range_column_storage_type = std::make_shared<DataTypeInt64>();
+    auto range_column_updated = castColumnAccurate(column_to_cast, range_column_storage_type);
```
Comment (nikitamikhaylov) [inline src/Dictionaries/RangeHashedDictionary.cpp:186]: Why cast?

Comment (kitaisreal) [reply src/Dictionaries/RangeHashedDictionary.cpp:186]: RangeHashed dictionary contains id and numeric representable range with size lower than 8 bytes, user fetch is based on id and this range, we cast range to Int64, to process it.

**Review by nikitamikhaylov — APPROVED** [review]

**Comment by olgarev** [comment]
@nikitamikhaylov  Hi! Does this really needs to be documented? 
See https://clickhouse.tech/docs/en/sql-reference/functions/ext-dict-functions/#dicthas
No list of supported dictionaries there.

**Comment by nikitamikhaylov** [comment]
> @nikitamikhaylov  Hi! Does this really needs to be documented? 
> 
> See https://clickhouse.tech/docs/en/sql-reference/functions/ext-dict-functions/#dicthas
> 
> No list of supported dictionaries there.

Let's ask @kitaisreal

**Comment by kitaisreal** [comment]
@olgarev no this feature is not needed to be documented.

**Comment by olgarev** [comment]
> @olgarev no this feature is not needed to be documented.

OK, thanks! Can you delete the "doc-alert" label then, please?

**Comment by kitaisreal** [comment]
@olgarev sure, done.

**Pull request merged**